### PR TITLE
KPI Feedback

### PR DESCRIPTION
**Pull request by gabrielcossette**
KPI ideas for IITB Services + Add PCRA for project intake

### CODE REVIEW

File: Strategies/IITB_KPIs.md [hunk 1]
```diff
@@ -30,13 +30,26 @@ Toil is the kind of work tied to running a production service that tends to be m
...
-`ratio = X / Y`
+We can measure both the service and development sides:
 
-2. For toil in ESDC Services, using the [ESDC's Service Catalogue](https://www.canada.ca/en/employment-social-development/corporate/portfolio/service-canada/programs.html) and measure the ratio of service requests to amount of personnel.
+For service requests (e.g. Helpdesk):
 
-`ratio = service requests / service personnel`
+`service ratio = service requests / service personnel`
+
+For the development side, there would be multiple metrics, some of them being influenced by the evolution toward DevOps, e.g.:
```
Comment (jaysonmc) [inline Strategies/IITB_KPIs.md]: I would suggest we use the standard Devops mertrics (we can review the annual DevOps reports by Dora and Puppet [the most popular]), but I also took notes [here](onenote:http://dialogue/grp/BU6810070/Shared%20Documents/IT%20Strategy%20Workbook/Notes.one#DORA%20-%20State%20of%20DevOps,%202018&section-id={8F7042FC-8CA2-4132-A942-B030364A6FE4}&page-id={A9CF60E1-957B-4421-A0CC-E2360B5B16D2}&end). About half way down you will see a grid (46 times more frequent code deployments, etc...) 

For development, sec & ops, I would suggest using these metrics

**Comment by gabrielcossette** [comment]
> 
> 
> Suggesting the use of standard DevOps metrics for development (including QA and security) metrics

Great suggestion @jaysonmc! I added metrics from the DORA report. You can review again to see if it makes sense, thanks!

**Comment by jaysonmc** [comment]
I would suggest removing some of the ones that are not as measurable, and add Gabriels suggestion (**PCRA**), and measuring the 4 types of work from the Phoenix Project and measure the **reduction of unplanned work**. 

Examples of ones we can remove ... 

7. Increase cyber resilience (I feel number 6 already addresses this in a more measurable way)
3. Reduction in toil of IITB and ESDC Services (important but not measureable. For example, we wrote 'ratio = service requests / service personnel' but ideally you want to reduce the number of service requests as well through automation. So the relationship between SRs and personnel is not linked, meaning the ratio here changing does not indicate a reduction in toil. That would only be true if we assume neither can change independently (which they can)
5 should result if 2 takes place ... do we need both? 
8. Client Satisfaction of IITB Services ... I assume business tracks this, so IITB shouldn't have to? Granted, it is a way we would measure our services. Just not sure if it belongs on the same list as 'vailability uptime of ESDC core systems'

File: Strategies/IITB_KPIs.md [hunk 2]
```diff
@@ -14,35 +14,77 @@ ESDC services provided to the public or to other departments will need to be del
...
+- `lead time for changes ratio =  time from code commit to running in production / dev personnel`
+- `test ratio = tests done / development personnel`
+- `*other?*`
+
+**Stability**
+- `time to restore ratio = time to restore service when incident occurs / dev personnel`
+- `change fail rate ratio = % of changes resulting in degraded service or remediation required / dev personnel`
+- `*other?*`
+
+**Efficiency**
```
Comment (remyb2canada) [inline Strategies/IITB_KPIs.md]: I worry we go too detailed in measurement. If we can stay strategic and influence other teams to achieve these strategic goals (e.g. ops team, testing team, governance with change approvals) we can let them measure how best to improve.
Recommendations:
1. Take out the manual work ratio and the categories underneath. We keep at the Throughput and Stability level only.
2. Time spent ratio is grouped under a "Quality Outcome" in the DORA report. Create a new KPI about Quality Work that involves the categories mentioned that also aligns with the Phoenix Project's 4 categories of work: business projects, IT projects (internal I&O), changes (bug fixes and security issues), unplanned work

From the DORA Accelerate report, this Efficiency section revolves around Quality.
This particular KPI invovles reducing toil

File: Strategies/IITB_KPIs.md [hunk 3]
```diff
@@ -14,35 +14,77 @@ ESDC services provided to the public or to other departments will need to be del
...
-`ratio = service requests / service personnel`
+`service ratio = service requests / service personnel`  
+
+For the development side, there would be multiple metrics, most of them being influenced by the evolution toward DevOps:
+
+**Throughput**
+- `release ratio = code deployments / dev personnel`
+- `lead time for changes ratio =  time from code commit to running in production / dev personnel`
+- `test ratio = tests done / development personnel`
+- `*other?*`
```
Comment (remyb2canada) [inline Strategies/IITB_KPIs.md]: take out *other*. If new ideas come in, new PR

File: Strategies/IITB_KPIs.md [hunk 4]
```diff
@@ -14,35 +14,77 @@ ESDC services provided to the public or to other departments will need to be del
...
+
+**Stability**
+- `time to restore ratio = time to restore service when incident occurs / dev personnel`
+- `change fail rate ratio = % of changes resulting in degraded service or remediation required / dev personnel`
+- `*other?*`
```
Comment (remyb2canada) [inline Strategies/IITB_KPIs.md]: take out *other*. If new ideas come in, new PR

File: Strategies/IITB_KPIs.md [hunk 5]
```diff
@@ -52,25 +94,27 @@ Rate of delivery is the measurement of IITB's ability to deliver projects and so
...
+1\. the rate at which intake capacity increases. The PMO's [Active Project List](http://dialogue/grp/ippst/Approved%20Projects%20Lists%20%28APL%29/Forms/AllItems.aspx?RootFolder=%2Fgrp%2Fippst%2FApproved%20Projects%20Lists%20%28APL%29%2FFY%202019%2D2020&FolderCTID=0x012000D8E2E348AB0A034886495AE4BED12E07&View=%7b2C779E42-EB28-444B-AACD-39CC61DDBCE2%7d), for Major/Minor/Small ESDC projects, as well as the PMO's [IITB Lite projects](http://dialogue/grp/ITPM-GPTI/Lists/IITBPMP%20Small%20Projects/AllItems.aspx) will be used as reference points to monitor intake increase. The project's complexity will be considered in the measurement.
```
Comment (remyb2canada) [inline Strategies/IITB_KPIs.md]: I sort of agree with @jaysonmc  :)
Remove 5. Incrase rate of delivery. Instead,  create a new KPI on Quality Work as per above comment. Measuring the 4 types of work mentioned in the Phoenix Project. The improvement metric here should be that IITB can focus more time on business project work, and less time on unplanned/changes.

**Comment by remyb2canada** [comment]
Before merging, we should apply the changes mentioned in this PR review and push it to the PR's commit.

then merge in master

File: Strategies/IITB_KPIs.md [hunk 6]
```diff
@@ -52,25 +94,27 @@ Rate of delivery is the measurement of IITB's ability to deliver projects and so
...
 
-`rate = (Incidents Year_n - Incidents Year_n-1) / Incidents Year_n-1`
+`rate of cyber incidents = (incidents year_n - incidents year_n-1) / incidents year_n-1`
 
 ## 7. **Increase cyber resilience**
 
 Cyber resilience is the ability to quickly mitigate and remediate a breach, whether it be from human error or insecure software resulting in operational disruption, data theft or public trust degradation. Cyber resilience will be measured by a decrease rate of intrusion, which is a complex measurement using security frameworks like the [Mitre Att&amp;ck Matrix](https://attack.mitre.org/).
 
-`rate = TBD`
+`rate of cyber resilience = TBD`
```
Comment (arushisharma19) [inline Strategies/IITB_KPIs.md]: Some thoughts on - 
Cyber resilience may be measured on “not losing operation capability” or “not losing protected information” in an event of cyber attack. 
Situational awareness about our systems could be part of cyber resilience.
From an article – “The challenge of Cyber Resilience can be expressed as a simple question: When security controls fail, can you detect unusual or irregular behaviour with sufficient context to mitigate the risk to the organisation?”
Cyber resilience can be how soon the operational capability was restored?

**Comment by gcharest** [comment]
Have all changes been reviewed/made? @gabrielcossette @remyb2canada @jaysonmc @arushisharma19 ?

We'll need to rebase. Thanks!

**Comment by gabrielcossette** [comment]
I just finished implementing @remyb2canada suggestions. Done for my side.
Regarding yours @jaysonmc, I think you want to push them through another PR?

**Comment by gabrielcossette** [comment]
But I'll need your collective review of my first stab at the new "5. Quality Work" KPI, after rebase I guess :)

File: Strategies/IITB_KPIs.md (hunk 6 above)
Comment (gabrielcossette) [reply Strategies/IITB_KPIs.md]: Great thoughts! Resilience could be measured during outage testing (e.g. if we adopt Chaos Engineering practice). I'll merge for now but feel free to submit a PR with proposed formula(s) and we'll discuss it :)

Comment (remyb2canada) [reply Strategies/IITB_KPIs.md]: Chaos Engineering. That may well be a practice that EO (enterprise operations) should be doing and funded for. Maybe even as part of IT Security program.

**Pull request merged**